eng/pipelines/templates/jobs/ci.yml

```diff
@@ -75,20 +75,19 @@ jobs:
           MatrixFilters: ${{ parameters.MatrixFilters }}
           MatrixReplace: ${{ parameters.MatrixReplace }}
           SparseCheckoutPaths: [ "sdk/", ".vscode"]
-          PreGenerationSteps:
-            - template: /eng/common/pipelines/templates/steps/save-package-properties.yml
-              parameters:
-                ServiceDirectory: ${{parameters.ServiceDirectory}}
+          ${{ if eq(parameters.ServiceDirectory, 'auto') }}:
+            EnablePRGeneration: true
+
+            PRMatrix:
+              Name: pr_test_base
+              Path: eng/pipelines/templates/stages/platform-matrix.json
+              Setting: ArtifactPackageNames
```

Comment:
I see, you made the naming dynamic because each language will have a different variable name for their package list?

Reply:
Exactly. It'll default to `ArtifactPackageNames` but whoever defines` PRMatrix` at call site can set it to whatever they need.